swap/swap_test.go

```diff
@@ -682,3 +686,246 @@ func testDeploy(ctx context.Context, backend cswap.Backend, swap *Swap) (err err
+	swap, dir := newTestSwap(t)
+	// owner address is the beneficary (counterparty) for the peer
+	// that's because we expect cheques we receive to be signed by the address we would issue cheques to
+	peer := NewPeer(newDummyPeer().Peer, swap, nil, ownerAddress, testChequeContract)
+	// we need to adjust the owner address on swap because we will issue cheques to beneficiaryAddress
+	swap.owner.address = beneficiaryAddress
+	return swap, peer, dir
+}
+
+// TestPeerSaveAndLoadLastReceivedCheque tests if a saved last received cheque can be loaded again later using the peer functions
+func TestPeerSaveAndLoadLastReceivedCheque(t *testing.T) {
+	_, peer, dir := newTestSwapAndPeer(t)
+	defer os.RemoveAll(dir)
+
+	testCheque := newTestCheque()
+
+	if err := peer.saveLastReceivedCheque(testCheque); err != nil {
+		t.Fatalf("Error while saving: %s", err.Error())
+	}
+
+	returnedCheque := peer.loadLastReceivedCheque()
+
+	if returnedCheque == nil {
+		t.Fatal("Could not find saved cheque")
+	}
+
+	if returnedCheque.Amount != testCheque.Amount || returnedCheque.Beneficiary != testCheque.Beneficiary {
+		t.Fatal("Returned cheque was different")
+	}
+}
+
+// TestPeerVerifyChequeProperties tests that verifyChequeProperties will accept a valid cheque
+func TestPeerVerifyChequeProperties(t *testing.T) {
+	_, peer, dir := newTestSwapAndPeer(t)
+	defer os.RemoveAll(dir)
+
+	testCheque := newTestCheque()
+	testCheque.Sig = testChequeSig
+
+	if err := peer.verifyChequeProperties(testCheque); err != nil {
+		t.Fatalf("failed to verify cheque properties: %s", err.Error())
+	}
+}
+
+// TestPeerVerifyChequeProperties tests that verifyChequeProperties will reject invalid cheques
+func TestPeerVerifyChequePropertiesInvalidCheque(t *testing.T) {
+	swap, peer, dir := newTestSwapAndPeer(t)
+	defer os.RemoveAll(dir)
+
+	// cheque with an invalid signature
+	testCheque := newTestCheque()
+	testCheque.Sig = manipulateSignature(testChequeSig)
+	if err := peer.verifyChequeProperties(testCheque); err == nil {
+		t.Fatalf("accepted cheque with invalid signature")
+	}
+
+	// cheque with wrong contract
+	testCheque = newTestCheque()
+	testCheque.Contract = beneficiaryAddress
+	testCheque.Sig, _ = swap.signContentWithKey(testCheque, ownerKey)
+	if err := peer.verifyChequeProperties(testCheque); err == nil {
+		t.Fatalf("accepted cheque with wrong contract")
+	}
+
+	// cheque with wrong beneficiary
+	testCheque = newTestCheque()
+	testCheque.Beneficiary = ownerAddress
+	testCheque.Sig, _ = swap.signContentWithKey(testCheque, ownerKey)
+	if err := peer.verifyChequeProperties(testCheque); err == nil {
+		t.Fatalf("accepted cheque with wrong beneficiary")
+	}
+
+	// cheque with non-zero timeout
+	testCheque = newTestCheque()
+	testCheque.Timeout = 10
+	testCheque.Sig, _ = swap.signContentWithKey(testCheque, ownerKey)
+	if err := peer.verifyChequeProperties(testCheque); err == nil {
+		t.Fatalf("accepted cheque with non-zero timeout")
+	}
```

Comment:
Do we need to add a test for a wrong amount if a cheque already exists and for the serial? The same if no cheque exists? Or is that covered by other tests